### PR TITLE
Add Suggest Loi Card View

### PR DESCRIPTION
**Pull request by JSunde**
Towards #1541

Adds a new card view for Suggest Loi cards and manages them alongside the Loi cards in the MapCardAdapter

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt [hunk 1]
```diff
@@ -22,32 +22,57 @@ import androidx.core.content.res.ResourcesCompat
...
+    } else {
+      SuggestLoiViewHolder(
+        SuggestLoiCardItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+      )
+    }
 
-  /** Binds [LocationOfInterest] data to [ViewHolder]. */
-  override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+  override fun getItemViewType(position: Int): Int =
+    if (position <= indexOfLastLoi) {
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt:53]: Is it possible to have more than one suggest loi type card?

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt [hunk 2]
```diff
@@ -22,32 +22,57 @@ import androidx.core.content.res.ResourcesCompat
...
 
-  /** Creates a new [ViewHolder] item without any data. */
-  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-    val binding = LoiCardItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
-    return ViewHolder(binding)
-  }
+  /** Creates a new [RecyclerView.ViewHolder] item without any data. */
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder =
+    if (viewType == R.layout.loi_card_item) {
+      LoiViewHolder(LoiCardItemBinding.inflate(LayoutInflater.from(parent.context), parent, false))
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt:45]: Can we pull the layoutInflater to a local var for better readability?

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt (hunk 1 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt:53]: Yes

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt [hunk 3]
```diff
@@ -22,32 +22,57 @@ import androidx.core.content.res.ResourcesCompat
...
+      R.layout.loi_card_item
+    } else {
+      R.layout.suggest_loi_card_item
+    }
+
+  /** Binds [LocationOfInterest] data to [LoiViewHolder] or [SuggestLoiViewHolder]. */
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
     val uiData: MapCardUiData = itemsList[position]
 
+    var loiCard: MaterialCardView? = null
```
[the remark this answers is not in the document]
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt:63]: Done

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt (hunk 1 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapCardAdapter.kt:53]: So every element after the last loi card will be a suggest loi card

**Pull request merged**